nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBRepository.java

```diff
@@ -1126,15 +1128,20 @@ public void clearLockManagerCaches() {
     }
 
     protected FindIterable<Document> find(Bson filter) {
-        return coll.find(filter).maxTime(maxTimeMS, MILLISECONDS);
+        return coll.find(filter).maxTime(getMaxTimeMs(), MILLISECONDS);
+    }
+
+    protected long getMaxTimeMs() {
+        long ttl = TransactionHelper.getTransactionTimeToLive();
+        return ttl < 0 ? maxTimeMS : ttl * 1000;
```

Comment:
For the `0` case the code in `com.mongodb.internal.operation.FindAndUpdateOperation.createCommand` will skip the parameter so use the server-side default. We should use maybe 1s in that case.

Reply:
let's add an extra half second all the time then

Reply:
That works for me. Could you document in `TransactionHelper.getTransactionTimeToLive` the meaning of `0` too btw?